homeassistant/helpers/script.py

```diff
@@ -294,16 +352,19 @@ async def async_run(self) -> None:
             self._finish()
 
     async def _async_step(self, log_exceptions):
-        with trace_path(str(self._step)), trace_action(self._variables):
-            try:
-                handler = f"_async_{cv.determine_script_action(self._action)}_step"
-                await getattr(self, handler)()
-            except Exception as ex:
-                if not isinstance(ex, (_StopScript, asyncio.CancelledError)) and (
-                    self._log_exceptions or log_exceptions
-                ):
-                    self._log_exception(ex)
-                raise
+        with trace_path(str(self._step)):
+            async with trace_action(self._hass, self, self._stop, self._variables):
+                if self._stop.is_set():
+                    return
+                try:
+                    handler = f"_async_{cv.determine_script_action(self._action)}_step"
+                    await getattr(self, handler)()
+                except Exception as ex:
+                    if not isinstance(ex, (_StopScript, asyncio.CancelledError)) and (
```

Comment:
`asyncio.CancelledError` inherits from `BaseException` now instead of `Exception` so I don't think we need that part of the check.

https://docs.python.org/3.8/library/asyncio-exceptions.html#asyncio.CancelledError

Reply:
Fixed in #47721